main/options.c

```diff
@@ -57,7 +57,7 @@
 #define EXTENSION_SEPARATOR '.'
 #define PATTERN_START '('
 #define PATTERN_STOP  ')'
-#define IGNORE_SEPARATORS   ", \t\n"
```

Comment:
man page said about -I
```
Otherwise, identifier-list is a list of identifiers (or identifier pairs) to be spe‐
            cially handled, each delimited by a either a comma or by white space (in which case the list should  be
            quoted  to keep the entire list as one command line argument).
```

whitespace is considered as a char for separation. Your new code may break the compatiblity.
Let's avoid it if possible.

My idea is that introducing `--opt-<LANG>=<PARAM>:<VALUE>` facility.
In this case the option will be `--opt-Cpreproc=define:AnythingYouWant`.